src/UI/templates/js/Button/button.js

```diff
@@ -0,0 +1,31 @@
+var il = il || {};
+il.UI = il.UI || {};
+
+(function($, il) {
+
+	$(function() {
+		$(".il-btn-month").each(function () {
+			var id = this.getAttribute("id");
+			$(this).find(".inline-picker").
+			datetimepicker({
+				inline: true,
+				sideBySide: true,
+				viewMode: 'months',
+				format: 'MM/YYYY'
+			}).on('dp.change', function (ev) {
+				var i, d , months = [];
+				var d = new Date(ev.date);
+				var m = d.getMonth() + 1;
+				m = ("00" + m).substring(m.toString().length);
+
+				for (i = 1; i<=12; i++) {
+					months.push(il.Language.txt("month_" + (("00" + i).substring(i.toString().length)) + "_long"));
+				}
+
+				$("#" + id + " span.il-current-month").html(months[d.getMonth()] + " " + d.getFullYear());
+				$("#" + id).trigger("il.ui.button.month.changed", [ id,  m + "-" + d.getFullYear()] );
+			});
+		});
+	});
+
+})($, il);
```

Comment:
> I did not want to expose any interface in this case. It is just an "iffe". No other consumer should be able to call il.UI.button.init()

oke.

> Additionally $tpl->addOnLoadCode("il.UI.button.init();"); produces Javascript tags in the page, which became an anti-pattern imo. E.g. it currently keeps us from implementing a simple CSP in ILIAS, see https://de.wikipedia.org/wiki/Content_Security_Policy

You have a point there. What about calling il.Util.addOnLoad directly from button.js then? I just like to throw all add on load code through the same pipeline to somehow.

> But, I see that this may lead to issues when trying to include this in asynchronous calls. Tricky. Are there similar discussions in other PRs?

Unfortunately not that I can think of. We still finding our way and each time we discuss JS, we get a little smarter. 

In this case, You can do what you think is best since I can not provide really strong arguments for one or the other at this point. I personally like the proposed way a little better since it is more in line with what is already there but you made two good points.